sound/soc/sof/intel/hda.c

```diff
@@ -1045,12 +1045,32 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 			 */
 			unique_id = SDW_UNIQUE_ID(adr);
 			if (link->num_adr == 1 ||
-			    ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID ||
 			    ids[j].id.unique_id == unique_id) {
 				dev_dbg(bus->dev,
 					"found %x at link %d\n",
 					part_id, link_id);
 				break;
+			} else if (ids[j].id.unique_id == SDW_IGNORED_UNIQUE_ID) {
+				unsigned int part_id2;
+
+				/* make sure there is no the same codec on the link */
+				for (k = 0; k < link->num_adr; k++) {
+					if (k == i)
+						continue;
+					part_id2 = SDW_PART_ID(link->adr_d[k].adr);
+					if (part_id2 == part_id) {
+						dev_dbg(bus->dev,
+							"Slave %x id %d not found\n",
```

Comment:
it's more  'wrong machine descriptor: link %d has duplicate part_id %d already tagged as unique'